code/modules/power/port_gen.dm

```diff
@@ -315,6 +315,10 @@ display round(lastgen) and plasmatank amount
 			usr << browse(null, "window=port_gen")
 			usr.unset_machine()
 
+/obj/machinery/power/port_gen/npc_tamper_act(mob/living/L)
+	active = !active
+	icon_state = "portgen[active]"
```

Comment:
Making this into an `update_icon` or anything more standardized would be a good idea.